daprdocs/content/en/reference/cli/dapr-annotate.md

```diff
@@ -0,0 +1,84 @@
+| `--http-stream-request-body` | | `false` | Enable streaming request body for HTTP |
+| `--listen-addresses` | | | The addresses for sidecar to listen on. To listen to all IPv4 addresses, use `0.0.0.0`. To listen to all IPv6 addresses, use `[::]`. |
+| `--liveness-probe-delay` | | `-1` | The delay for sidecar to use for the liveness probe. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--liveness-probe-period` | | `-1` | The period used by the sidecar for the liveness probe. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--liveness-probe-threshold` | | `-1` | The threshold used by the sidecar for the liveness probe. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--liveness-probe-timeout` | | `-1` | The timeout used by the sidecar for the liveness probe. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--log-level` | | | The log level to use |
+| `--max-request-body-size` | | `-1` | The maximum request body size to use |
+| `--memory-limit` | | | The memory limit to set for the sidecar. See valid values [here](https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/) |
+| `--memory-request`| | | The memory request to set for the sidecar |
+| `--metrics-port` | | `-1` | The port to expose the metrics on |
+| `--namespace, -n` | | | The namespace the resource target is in (can only be set if `--resource` is also set) |
+| `--readiness-probe-delay` | | `-1` | The delay to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes).|
+| `--readiness-probe-period` | | `-1` | The period to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--readiness-probe-threshold` | | `-1` | The threshold to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--readiness-probe-timeout` | | `-1` | The timeout to use for the readiness probe in the sidecar. Read more [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes). |
+| `--resource, -r` | | | The resource to target to annotate |
+
+{{% alert title="Warning" color="warning" %}}
+If an application ID is not provided using `--app-id, -a`, we will generate one using the format `<namespace>-<kind>-<name>`.
```

Comment:
app-id is mandatory, however, you may be modifying an existing manifest that already has an app-id annotation - for instance `dapr annotate --log-level debug https://raw.githubusercontent.com/dapr/quickstarts/master/tutorials/hello-kubernetes/deploy/node.yaml` - this is just changing the log level of an existing dapr app. Rather than having to overwrite the app-id every time I just made it optional with a default convention if not provided and not already present. If you'd rather we didn't use this convention then I can make app-id required. Let me know.